C++/source/number_theory/gcd.cpp

```diff
@@ -19,11 +19,15 @@
 #include <iostream>
 using namespace std;
 
+// Function to return maximum/minimum of a and b
+int max(int a, int b) { a>b ? a : b; }
+int min(int a, int b) { a<b ? a : b; }
+
 // Function to return gcd of a and b
 int gcd(int a, int b) {
-    if (a == 0)
-        return b;
-    return gcd(b%a, a);
+    if (min(a, b) == 0)
+        return max(a, b);
+    return gcd(max(a, b) % min(a, b), min(a, b));
```

Comment:
The existing code already takes care of what you're trying to add.

Consider this example case:
`gcd(152, 74)` :arrow_right: `gcd(74, 152)` :arrow_right: `gcd(4, 74)` :arrow_right: `gcd(2, 4)` :arrow_right: `gcd(0, 2)` :arrow_right: `2`

Whether you call `gcd(larger_num, smaller_num)` or `gcd(smaller_num, larger_num)`, the result will be the same.